Use an inline namespace for Abseil.

This prevents Drake's build of Abseil from interfering with a user's
downstream build of Abseil when the user is using static linking.

--- absl/base/options.h.orig	2022-03-08 13:01:16.000000000 -0800
+++ absl/base/options.h	2022-03-13 13:44:21.385254191 -0700
@@ -205,8 +205,8 @@
 // be changed to a new, unique identifier name.  In particular "head" is not
 // allowed.
 
-#define ABSL_OPTION_USE_INLINE_NAMESPACE 0
-#define ABSL_OPTION_INLINE_NAMESPACE_NAME head
+#define ABSL_OPTION_USE_INLINE_NAMESPACE 1
+#define ABSL_OPTION_INLINE_NAMESPACE_NAME drake_vendor
 
 // ABSL_OPTION_HARDENED
 //
